src/context/MyProvider.js

```diff
@@ -70,109 +71,125 @@ class MyProvider extends Component {
 					Authorization: this.token,
 				},
 			})
-				.then((data) => data.json())
 				.then((data) => {
-					const files = data.searches[0].files.filter((file) => {
-						file.time = this.ConvertDate(file.timestamps.updatedAt);
-						return file;
-					});
-					console.log("File,", data);
-					this.setState({
-						Loaded: true,
-						documents: [...files],
-					});
-					this.backup = [...files];
+					if (data.status === 200) {
+						status = 200;
+						return data.json();
+					} else {
+						status = 500;
+						localStorage.removeItem("token");
+						console.log("Loading Failed");
+						console.log(data.json);
+						window.location.reload(false);
+						return data.json();
+					}
+				})
+				.then((data) => {
+					console.log(status);
+					if (status === 200) {
+						const files = data.searches[0].files.filter((file) => {
+							file.time = this.ConvertDate(
+								file.timestamps.updatedAt
+							);
+							return file;
+						});
+						this.setState({
+							Loaded: true,
+							documents: [...files],
+						});
+						this.backup = [...files];
+					}
 				});
 		}
 	};
 	NewDocumentHandler = () => {
-		if (this.guest !== "Yes") {
-			return new Promise((resolve, reject) => {
-				fetch(this.apiUrl + "preview/createFile/", {
-					method: "PATCH",
-					headers: {
-						Authorization: this.token,
-						"Content-Type": "application/json",
-					},
-					body: JSON.stringify({
-						fileName: "New Document",
-						userId: this.userId,
-					}),
-				})
-					.then((data) => data.json())
-					.then((data) => {
-						data.created.time = this.ConvertDate(
-							data.created.timestamps.updatedAt
-						);
-						this.setState({
-							documents: [...this.state.documents, data.created],
-						});
-						resolve(data.created.fileId);
+		// if (this.guest !== "Yes") {
+		return new Promise((resolve, reject) => {
+			fetch(this.apiUrl + "preview/createFile/", {
+				method: "PATCH",
+				headers: {
+					Authorization: this.token,
+					"Content-Type": "application/json",
+				},
+				body: JSON.stringify({
+					fileName: "New Document",
+					userId: this.userId,
```

Comment:
Why is the userId being sent in the request payload?

Reply:
Removed